src/compiler/output-targets/dist-custom-elements/custom-elements-types.ts

```diff
@@ -54,16 +59,20 @@ const generateCustomElementsTypesOutput = async (
     ...components.map((component) => {
       const exportName = dashToPascalCase(component.tagName);
       const importName = component.componentClassName;
-      // typedefs for individual components can be found under paths like
-      // $TYPES_DIR/components/my-component/my-component.d.ts
-      //
-      // To construct this path we:
-      //
-      // - get the relative path to the component's source file from the source directory
-      // - join that relative path to the relative path from the `index.d.ts` file to the
-      //   directory where typedefs are saved
-      const componentSourceRelPath = relative(config.srcDir, component.sourceFilePath).replace('.tsx', '');
-      const componentDTSPath = join(componentsTypeDirectoryRelPath, componentSourceRelPath);
+
+      let componentDTSPath = join(componentsTypeDirectoryRelPath, component.tagName, component.tagName);
```

Comment:
ATM, `isBarrelExport` is falsey IFF (if & only if) `outputTarget.customElementsExportBehavior === 'default'`. In the scenario for 'default', where we don't generate any additional types, and use the `componentDTSPath` calculated here. However, I don't believe this path is correct - by:
1. Spinning up a new project with `npm init stencil component`
2. Installing this branch in the starter project 
3. Adding the following to `stencil.config.ts`:
```diff
diff --git a/stencil.config.ts b/stencil.config.ts
index b8fb12f..76697ff 100644
--- a/stencil.config.ts
+++ b/stencil.config.ts
@@ -9,7 +9,8 @@ export const config: Config = {
     },
     {
       type: 'dist-custom-elements',
+      customElementsExportBehavior: 'default',
+      generateTypeDeclarations: true,
     },
     {
```
4. Renaming the default `MyComponent` to `MyComponentClazz` and the tag to `my-component-diff-tag`
4. Running `npm run build` yields a file with an import path like so:
```ts
// dist/components/index.d.ts
export { MyComponentClazz as MyComponentDiffTag } from '../types/components/my-component-diff-tag/my-component-diff-tag';
```
However, that path in the export statement does not exist on the filesystem.

I think this is occurring because in v2.X, where the original changes from https://github.com/ionic-team/stencil/pull/3468/files occurred, we don't create types be default. However in v3.X, we do generate those types https://github.com/ionic-team/stencil/pull/3502/files#diff-4a8e48905c6379413a82ea0524130976379f57f8976e2abdfdb44082c0fc0499R40.  

I think we need to do a little more here to ensure type generation is taken into account - can you PTAL?

Reply:
Think I got this squared away. Now, these paths will only be generated if the `single-export-module` behavior is set and uses the correct logic to generate the paths (I initially reverted behavior _too_ far which is why you were seeing paths that didn't exist). Tested the output with combinations of type generation and `customElementsExportBehavior` we currently have.